File: tests/Generator/resource/python/complex/security_api_key.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union, Literal
from .security import Security


class SecurityApiKey(Security):
    type: Literal["apiKey"] = Field(alias="type")
    in_: Optional[str] = Field(default=None, alias="in")
    name: Optional[str] = Field(default=None, alias="name")
    pass

File: tests/Generator/resource/rust/complex/security_api_key.rs
use serde::{Serialize, Deserialize};
use security::Security;

#[derive(Serialize, Deserialize)]
pub struct SecurityApiKey {
    #[serde(rename = "type")]
    _type: Option<String>,

    #[serde(rename = "in")]
    _in: Option<String>,

    #[serde(rename = "name")]
    name: Option<String>,

}

